src/emscripten-optimizer/optimizer-shared.cpp

```diff
@@ -67,7 +56,7 @@ AsmType detectType(Ref node, AsmData *asmData, bool inVarDef, IString minifiedFr
   switch (node[0]->getCString()[0]) {
     case 'n': {
       if (node[0] == NUM) {
-        if (!isInteger(node[1]->getNumber())) return ASM_DOUBLE;
+        if (!wasm::isInteger(node[1]->getNumber())) return ASM_DOUBLE;
```

Comment:
the difference is that emscripten_optimizer is also used in emscripten, where it powers the asm.js optimizer. it happens to be useful in binaryen for asm2wasm and wasm2asm, etc., so it's here as well.

perhaps it could be moved out to a repo of it's own, but that seems kind of overkill.

btw, in emscripten the license is MIT - are you ok with dual-licensing your changes, so that they can go there as well?